fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java

```diff
@@ -226,7 +225,6 @@ private void collectAllPredicates(OptExpression root) {
         if (logicalOperator.getPredicate() != null) {
             ScalarOperator scalarOperator = logicalOperator.getPredicate();
             if (!scalarOperator.isConstantRef()) {
-                Preconditions.checkState(scalarOperator instanceof PredicateOperator);
```

Comment:
What the actual class is accroding to your unit test?

Reply:
CallOperator

Reply:
@kangkaisen 
why not add checkState (scalarOperator instanceof CallOperator)